### PR TITLE
Gzipped and non-utf8 content support

### PR DESCRIPTION
**Pull request by sdrobov**
I added auto-detection and handling of gzipped content and content in non-unt8 encoding. I took the code from [feedparser](https://github.com/danmactough/node-feedparser/blob/master/examples/compressed.js) [example](https://github.com/danmactough/node-feedparser/blob/master/examples/iconv.js) and adapted it.

### CODE REVIEW

**Comment by alabeduarte** [comment]
Hi @sdrobov 

Thanks for the PR!

I noticed you added some extra behaviour to the parsing, would you mind to add some level of automated tests for them? You can check a few examples on this file https://github.com/alabeduarte/feedparser-promised/blob/master/test/feedParserPromised.test.js

Cheers

File: src/feedParserPromised.js [hunk 1]
```diff
@@ -2,27 +2,85 @@
...
+let items = [];
+let feedparser;
 
 module.exports = class FeedParserPromised {
   static parse (requestOptions, feedparserOptions) {
-    return new Promise( (resolve, reject) => {
-      const items = [];
-      const feedparser = new FeedParser(feedparserOptions);
+    return new Promise((resolve, reject) => {
+      requestOptions = typeof requestOptions == 'string' ? { uri: requestOptions, onError: reject } : requestOptions;
```
Comment (alabeduarte) [inline src/feedParserPromised.js:14]: Have you considered creating some unit tests for this logic? 

Suggestion: maybe it's an opportunity to extract this into a different abstraction like `RequestParser` that could be injected into this function, which would also be easier to test.

I'd suggest doing the same for `_maybeDecompress` and `_maybeTranslate`.

**Comment by alabeduarte** [comment]
Hey @sdrobov 

Any thoughts regarding unit tests?

**Comment by sdrobov** [comment]
Hi @alabeduarte
Oh, sorry, I miss your comment, was busy at work. Sure I will create tests in 3-5 days.

**Comment by sdrobov** [comment]
@alabeduarte here is the tests. sorry it takes so long, i was really very busy

**Comment by alabeduarte** [comment]
Hi @sdrobov 

Thanks for sending the changes!

Some of the dependencies were considered compromised for the security standpoint according to the GitHub report and I worked on some upgrades and released the version number 2.0.0 this weekend. I took the opportunity to also split the code a bit removing what is `FeedParser` concern and what is HTTP request. Do you mind to have a look at where this PR fits on the new structure? I know that it might be a lot of changes... We can work together on this if you find trouble.

Cheers

**Comment by alabeduarte** [comment]
Hi @sdrobov 

I'm closing the PR since I didn't receive a reply for a while, but let me know if you still are interested to contribute. Basically, all we have to do is to solve the conflicts, but I'm sure the tests will back you up.

Cheers

**Pull request closed without merging**